src/Datastore/DatastoreSessionHandler.php

```diff
@@ -0,0 +1,277 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Datastore;
+
+use Exception;
+use InvalidArgumentException;
+use SessionHandlerInterface;
+
+/**
+ * Custom session handler backed by Cloud Datastore.
+ *
+ * {@see http://php.net/manual/en/class.sessionhandlerinterface.php}
+ *
+ * Instead of storing the session data in a local file, it stores the data to
+ * Cloud Datastore. The biggest benefit of doing this is the data can be
+ * shared by multiple instances, so it's suitable for cloud applications.
+ *
+ * The downside of using Cloud Datastore is the write operations will cost you
+ * some money, so it is highly recommended to minimize the write operations
+ * with your session data with this handler. In order to do so, keep the data
+ * in the session as limited as possible; for example, it is ok to put only
+ * signed-in state and the user id in the session with this handler. However,
+ * for example, it is definitely not recommended that you store your
+ * application's whole undo history in the session, because every user
+ * operations will cause the Datastore write and then it will cost you lot of
+ * money.
+ *
+ * This handler doesn't provide pessimistic lock for session data. Instead, it
+ * uses Datastore Transaction for data consistency. This means that if
+ * multiple requests are modifying the same session data simultaneously, there
+ * will be more probablity that some of the `write` operations will fail.
+ *
+ * If you are building an ajax application which may issue multiple requests
+ * to the server, please design the session data carefully, in order to avoid
+ * possible data contentions. Also please see the 2nd example below for how to
+ * properly handle errors on `write` operations.
+ *
+ * It uses the session.save_path as the Datastore namespace for isolating the
+ * session data from your application data, it also uses the session.name as
+ * the Datastore kind, the session id as the Datastore id. By default, it
+ * does nothing on gc for reducing the cost. Pass positive value up to 1000
+ * for $gcLimit parameter to delete entities in gc.
+ *
+ * Note: The datastore transaction only lasts 60 seconds. If this handler is
+ * used for long running requests, it will fail on `write`.
+ *
+ * Example without error handling:
+ * ```
+ * use Google\Cloud\Datastore\DatastoreClient;
```

Comment:
What's the benefit of using the `ServiceBuilder`?

Actually on our public documentation like [the pubsub page](https://cloud.google.com/pubsub/docs/reference/libraries#php-usage), we don't advertise the ServiceBuilder because it is way simpler.

I can change it though, but I'd like to know the rationale of advertising the `ServiceBuilder`.